### ISSUE TITLE
My Whole UI project got deleted 😩🤦‍♂️

### ISSUE THREAD

**Issue by anandchakru** [node-gradle/gradle-node-plugin#297]
Please **DO NOT DELETE** everything in any directory, no matter what configuration/misconfiguration is in build.gradle file. Here is what I had in my build.gradle when this DISASTER happened 😩.

```
node {
	version = '20.10.0'
	download = true
	workDir = file("${project.projectDir}/my-ui") # should have been nodeProjectDir (instead of workDir)
}

task installNpm(type: NodeTask) {
	dependsOn npmInstall
	script = file("${project.projectDir}/my-ui/build.js")
}

task buildReactUI(type: NodeTask) {
	script = file("${project.projectDir}/my-ui/build.js")
	args = ["run", "build"]
}

task copyReactUI(type: Copy) {
	dependsOn buildReactUI
	from "${project.projectDir}/my-ui/build"
	into "${project.projectDir}/src/main/resources/static/public"
}

processResources {
	dependsOn copyReactUI
}
```

After I ran gradle clean build, this is what I have in my-ui directory
`drwxr-xr-x@  9 loki  staff  288 Dec 24 22:24 node-v20.10.0-darwin-x64`

**Comment by deepy** [comment]
I'm so sorry that this happened, this is a decision that was made before the project was forked and something we probably should have scrutinized much earlier
The reason this cleanup is in place is to prevent node installations from accumulating and filling up the disk, so while the functionality is probably not going away we _absolutely_ should check that we're only deleting node

**Comment by deepy** [comment]
This has been fixed in 7.0.2, we now only delete files that either start with `node-v` or are in a folder starting with `node-v`
It's not foolproof but should help prevent worst-case scenarios
